File: 14day/source.java
import java.util.Scanner;
import java.io.FileInputStream;
public class source {
	
	private static int[][] road = new int[11][11];
	private static int n, m;
	private static int start, end, value;
	private static int sum;
	private static int min = 0x7fffffff;
	private static int[] chk = new int[11];
	
	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(new FileInputStream("sample.txt"));	

		n = sc.nextInt();
		m = sc.nextInt();

		for(int i = 0; i < m; i++){
			start = sc.nextInt();
			end = sc.nextInt();
			value = sc.nextInt();
			road[start][end] = road[end][start] = value;
		}
		solve(1);

		System.out.println(min==0x7fffffff ? "‐1" : min);
	}
	
	private static void solve(int a){
		if(a == n){
			if(min > sum) min = sum;
			return;
		}
		
		for(int i = 2; i <= n; i++)
			if(road[a][i] != 0 && chk[i] == 0){
				sum += road[a][i]; chk[i] = 1;
				solve(i);
				sum -= road[a][i]; chk[i] = 0;
			}
	}
}

File: 17day/source.cpp
#define _CRT_SECURE_NO_WARNINGS //scanf 오류 제거
#include <stdio.h>

int n;
int min = 0x7fffffff;
int arr[10][10];
int chk[10];

void solve(int a, int sum) {
	if (a == n) {							//n개를 모두 탐색했다면
		if (min > sum)
			min = sum;
		return;
	}
	for (int j = 0; j < n; j++) {
		if (chk[j] == 0) {					//체크하지 않은 열이라면
			chk[j] = 1;						//체크
			solve(a + 1, sum + arr[a][j]);	//진입
			chk[j] = 0;						//체크해제
		}
	}
}

int main(){
	freopen("sample.txt", "r", stdin);

	scanf("%d", &n);

	for(int i = 0; i < n; i++)
		for(int j = 0; j < n; j++)
			scanf("%d", &arr[i][j]);

	solve(0, 0);

	printf("%d\n", min);
}